pilot/pkg/security/model/authentication.go

```diff
@@ -210,6 +213,80 @@ func AppendURIPrefixToTrustDomain(trustDomainAliases []string) []string {
 	return res
 }
 
+var fipsCiphers = []string{
+	"ECDHE-ECDSA-AES128-GCM-SHA256",
+	"ECDHE-RSA-AES128-GCM-SHA256",
+	"ECDHE-ECDSA-AES256-GCM-SHA384",
+	"ECDHE-RSA-AES256-GCM-SHA384",
+}
+
+var fipsGoCiphers = []uint16{
+	gotls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+	gotls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+	gotls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+	gotls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+}
+
+func index(ciphers []string) map[string]struct{} {
+	out := make(map[string]struct{})
+	for _, cipher := range ciphers {
+		out[cipher] = struct{}{}
+	}
+	return out
+}
+
+var fipsCipherIndex = index(fipsCiphers)
+
+// EnforceGoCompliance limits the TLS settings to the compliant values.
+// This should be called as the last policy.
+func EnforceGoCompliance(ctx *gotls.Config) {
+	switch common_features.CompliancePolicy {
+	case "":
+		return
+	case common_features.FIPS_140_2:
```

Comment:
nit: some link to where these requirements is defined may be helpful. Especially the no TLS1.3 -- I know why, but may be useful as a comment in code